lib/provisioning/Provisioner.js

```diff
@@ -341,6 +359,123 @@ Provisioner.prototype.handlePm = function(server, fromUser, text) {
+        //      info: 'some information' ,
+        //      error: 'an error'
+        // }
+        parameter_info: {
+            remote_room_server: {},
+            remote_room_channel: {},
+            matrix_room_id: {},
+            op_nick: {},
+            user_id: {}
+        },
+        // General info and errors
+        infos: [],
+        errors: [],
+        // List of operators. Only defined if retrieved.
+        // operators: []
+    }
+
+    try {
+        this._linkValidator.validate(options);
+    }
+    catch (err) {
+        // .validate does not return any details of problems with parameters
+        console.log(err);
+        queryInfo.errors.push('Parameter(s) malformed');
+        return queryInfo;
+    }
+
+    // Try to find the domain requested for linking
+    //TODO: ircDomain might include protocol, i.e. irc://irc.freenode.net
+    let server = this._ircBridge.getServer(ircDomain);
+
+    if (!server) {
+        queryInfo.parameter_info['remote_room_server']
+                 .error = 'Server not found';
+        return queryInfo;
+    }
+
+    if (server.isExcludedChannel(ircChannel)) {
+        queryInfo.parameter_info['remote_room_channel']
+                 .error = 'Server is configured to exclude channel';
+    }
+
+    let entry = yield this._ircBridge.getStore().getRoom(roomId, ircDomain, ircChannel);
+    if (entry) {
+        queryInfo.errors.push('Mapping already exists');
+    }
+
+    // Errors with nick prevent further querying
+    if (queryInfo.parameter_info['op_nick'].error) {
+        return queryInfo;
+    }
+
+    let existing = this._getRequest(server, opNick);
+    if (existing) {
+        let from = existing.userId;
+        queryInfo.errors.push(`Bridging request already sent to this op from ${from}`);
+    }
+
+    let hasPower = null;
+    try {
+        hasPower = yield this._userHasProvisioningPower(userId, roomId);
+    }
+    catch (err) {
+        log.error(err.stack);
+        queryInfo.errors.push('Could not retrieve power levels');
+    }
+
+    if (!hasPower) {
+        queryInfo.errors.push('You do not possess enough power to bridge this room');
+    }
+
+    // There must be no error with the channel to continue
+    if (queryInfo.parameter_info['remote_room_channel'].error) {
+        return queryInfo;
+    }
+
+    let botClient = yield this._ircBridge.getBotClient(server);
+
+    try {
+        yield botClient.joinChannel(ircChannel, key);
+    }
+    catch (err) {
+        log.error(err.stack);
+        queryInfo.parameter_info['remote_room_channel'].error = 'Bot cannot join this channel';
+        return queryInfo;
+    }
+
+    let opsInfo = yield botClient.getOperators(ircChannel);
+
+    queryInfo.operators = opsInfo.operatorNicks;
+
+    if (opsInfo.operatorNicks.indexOf(opNick) == -1) {
+        queryInfo.parameter_info['op_nick'].error = 'Provided user is not a chan op in this '+
+                                                    'channel';
```

Comment:
But the theory goes that a channel is always provided with the API call (or validation will fail), so they already know the channel, which is why all info and errors returned don't reference the room, channel, server, etc.